sqrl-engines/sqrl-engine-flink/src/main/java/com/datasqrl/schema/converters/UniversalTable2FlinkSchema.java

```diff
@@ -36,27 +38,48 @@ public DataType nullable(DataType type, boolean nullable) {
     }
   }
 
-  @Override
-  public DataType wrapArray(DataType type) {
+  private DataType wrapArray(DataType type) {
     return DataTypes.ARRAY(type);
   }
 
-  @Override
-  public DataType nestedTable(List<Pair<String, DataType>> columns) {
-    DataTypes.Field[] fields = new DataTypes.Field[columns.size()];
+  private DataType nestedTable(List<RelDataTypeField> relation) {
+    DataTypes.Field[] fields = new DataTypes.Field[relation.size()];
     int i = 0;
-    for (Pair<String, DataType> column : columns) {
-      fields[i++] = DataTypes.FIELD(column.getKey(), column.getValue());
+    for (RelDataTypeField column : relation) {
+      fields[i++] = DataTypes.FIELD(column.getName(), convertRelDataType(column.getType()));
     }
     return DataTypes.ROW(fields);
   }
 
+  private DataType convertRelDataType(RelDataType type) {
```

Comment:
I think the code in convertPrimitive will convert Row types but I'm not sure to what extent.

Reply:
yes, that seems to work. Removing the explicit conversion code.